### PR TITLE
[MIG] [16.0] `delivery_line_sale_line_position`: Migration to 16.0

### PR DESCRIPTION
**Pull request by ivantodorovich**
Depends on:
- https://github.com/OCA/sale-reporting/pull/211

### CODE REVIEW

**Comment by ivantodorovich** [comment]
pre-commit seems to be failing due to : 
- https://github.com/OCA/stock-logistics-reporting/pull/262

**Comment by simahawk** [comment]
LG

**Comment by simahawk** [comment]
Base PR is getting merged

File: delivery_line_sale_line_position/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+This module is build on top of the module `sale_order_line_position`.
+
+On the delivery report it adds a column with the `Postion` from the related
+sale order line.
```
Comment (imlopes) [inline delivery_line_sale_line_position/readme/DESCRIPTION.rst]:
```suggestion
On the delivery report it adds a column with the `Position` from the related
sale order line.
```

File: delivery_line_sale_line_position/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2021 Camptocamp SA
```
Comment (imlopes) [inline delivery_line_sale_line_position/__manifest__.py:1]:
```suggestion
# Copyright 2023 Camptocamp SA
```

File: delivery_line_sale_line_position/models/stock_move.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2021 Camptocamp SA
```
Comment (imlopes) [inline delivery_line_sale_line_position/models/stock_move.py:1]:
```suggestion
# Copyright 2023 Camptocamp SA
```

File: delivery_line_sale_line_position/models/stock_move_line.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2021 Camptocamp SA
```
Comment (imlopes) [inline delivery_line_sale_line_position/models/stock_move_line.py:1]:
```suggestion
# Copyright 2023 Camptocamp SA
```

**Review by imlopes — APPROVED** [review]
LGTM

File: delivery_line_sale_line_position/report/report_deliveryslip.xml [hunk 5]
```diff
@@ -26,18 +23,28 @@
...
+            position="before"
+        >
+            <th t-if="has_line_position and has_serial_number" name="th_sml_pos"><strong
+                >Pos</strong></th>
         </xpath>
+        <!--
+            NOTE: The rows are rendered by other templates.
+            `stock_report_delivery_has_serial_move_line` if the lines have lots/serials
+            `stock_report_delivery_aggregated_move_lines` if they don't. In this case the lines are
+            aggregated, and so we don't display the position.
```
Comment (sebalix) [inline delivery_line_sale_line_position/report/report_deliveryslip.xml:37]: Then should we not display the positions the line is referring to? Or avoid this aggregation done by Odoo.

File: delivery_line_sale_line_position/__manifest__.py (hunk 2 above)
Comment (simahawk) [reply delivery_line_sale_line_position/__manifest__.py:1]: the year should always be the 1st one. If not specified the "end" is always "today".

Comment (imlopes) [reply delivery_line_sale_line_position/__manifest__.py:1]: Thanks @simahawk 
I didn't know it

File: delivery_line_sale_line_position/report/report_deliveryslip.xml (hunk 5 above)
Comment (ivantodorovich) [reply delivery_line_sale_line_position/report/report_deliveryslip.xml:37]: Honestly I'm sure. In the uncertainty I decided to keep the same 15.0 behaviour + this note.

File: delivery_line_sale_line_position/report/report_deliveryslip.xml [hunk 6]
```diff
@@ -26,18 +23,28 @@
...
+        -->
     </template>
 
     <template
         id="stock_report_delivery_has_serial_move_line"
         inherit_id="stock.stock_report_delivery_has_serial_move_line"
     >
-        <xpath expr="//span[@t-field='move_line.product_id']/.." position="before">
+        <xpath expr="//td[1]" position="before">
```
Comment (simahawk) [inline delivery_line_sale_line_position/report/report_deliveryslip.xml:45]: the former selector seems more precise than this

Comment (ivantodorovich) [reply delivery_line_sale_line_position/report/report_deliveryslip.xml:45]: Yes but the point is not to be anchored to the `move_line.product_id` column, but rather to the first one (whatever it is).
That is, we don't care what the existing columns are, we want ours to be the first.

**Comment by simahawk** [comment]
@sebalix fine for you?

**Review by sebalix — APPROVED** [review]
Fine for me!

**Pull request merged**